exonum/src/runtime/rust/service.rs

```diff
@@ -137,6 +138,43 @@ where
     }
 }
 
+/// Simplifies creation of supplementary entities for `ServiceFactory`.
+pub trait InstanceInfoProvider: ServiceFactory {
+    /// Produces service instance configuration parameters based on service's artifact.
+    fn get_instance(
+        &self,
+        id: InstanceId,
+        name: impl Into<String>,
+        constructor: impl BinaryValue,
+    ) -> InstanceInitParams {
+        InstanceInitParams::new(id, name.into(), self.get_artifact(), constructor)
+    }
+
+    /// Returns service's ArtifactId.
+    fn get_artifact(&self) -> ArtifactId {
+        self.artifact_id().clone().into()
```

Comment:
Why not to use `artifact_id().clone()` directly?

Reply:
It's used by the default `DefaultInstance` implementation and manually in tests.